sap/rfc/user.py

```diff
@@ -312,13 +413,35 @@ def create_user(self, connection, user_builder: UserBuilder) -> BAPIReturn:
         """Creates a new user for the given user data"""
 
         rfc_ret = self._call_bapi_method(connection, 'BAPI_USER_CREATE1', user_builder.build_rfc_params())
-        return BAPIReturn(rfc_ret['RETURN'])
+        bapi_messages = rfc_ret['RETURN']
+        bapi_result = BAPIReturn(bapi_messages)
+        if bapi_result.is_error:
+            return bapi_result
+
+        rfc_password_change_ret = self._change_user_password(connection, user_builder)
```

Comment:
Why do we call it even if "productive" password was not requested?

Reply:
Yes, we are calling the method _change_user_password(). However, in it there is a check if we need the productive password logic or not. If we do not need it, then the method ends. 

I decided to do it this way, so that all the calls for user_password_manager are together in one method. I did not want to have one part in the method create_user() and the other part in the method _change_user_password().